### PR TITLE
Adding release make tag

### PR DESCRIPTION
**Pull request by xibz**
This will build a release binary that will trim out unwanted paths in
the binary. This can be done by setting the TRIMPATH environment
variable.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -15,8 +15,28 @@ SRCFILES := *.go
...
-	go build
+	go build -o firectl
 
+release: $(SRCFILES)
+	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build \
+		    -a \
+		    -installsuffix cgo \
+		    -ldflags "-s" \
+		    -gcflags=all=-trimpath=${TRIMPATH} \
+		    -asmflags=all=-trimpath=${TRIMPATH} \
```
Comment (mxpv) [inline Makefile]: Can you give some context why is this needed?

Comment (xibz) [reply Makefile]: Ah, basically if you don't include that you'll have your machines paths on there. For example, if you do `strings firectl | grep /home/mxpv` it'd spit out a bunch of paths. So to generate binaries with no knowledge of what host it is being built on, you have to strip them during compilation.

File: Makefile [hunk 2]
```diff
@@ -15,8 +15,28 @@ SRCFILES := *.go
...
 firectl: $(SRCFILES)
-	go build
+	go build -o firectl
 
+release: $(SRCFILES)
+	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build \
+		    -a \
+		    -installsuffix cgo \
+		    -ldflags "-s" \
+		    -gcflags=all=-trimpath=${TRIMPATH} \
```
Comment (samuelkarp) [inline Makefile]: I don't see `TRIMPATH` defined anywhere in the Makefile.  Do you want the build to fail if it's not defined?  You can do that by adding `test -z $(TRIMPATH)` above the `go build` line.

File: Makefile (hunk 1 above)
Comment (samuelkarp) [reply Makefile]: What we've done in other projects (like the ECS agent) is to define our release process such that builds always happen inside a particular, well-defined Docker container with a predictable path that we don't particularly care about.  I don't think trimpath is strictly necessary if we do that, and I would like us to use a well-defined Docker container for our builds so we can have as much of a reproducible environment as possible.

File: Makefile [hunk 3]
```diff
@@ -15,8 +15,28 @@ SRCFILES := *.go
...
+		    -o firectl
+
+sandbox-release-build:
+	docker build -f sandbox/Dockerfile -t "firectl" .
+
+sandbox-release-run:
+	docker run \
+		--init \
+		--rm \
+		-v ${MOUNTPATH}:/release \
```
Comment (samuelkarp) [inline Makefile]: This isn't defined anywhere.  What is it?

File: Makefile [hunk 4]
```diff
@@ -15,8 +15,28 @@ SRCFILES := *.go
...
+release: $(SRCFILES)
+	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build \
+		    -a \
+		    -installsuffix cgo \
+		    -ldflags "-s" \
+		    -gcflags=all=-trimpath=${TRIMPATH} \
+		    -asmflags=all=-trimpath=${TRIMPATH} \
+		    -o firectl
+
+sandbox-release-build:
```
Comment (samuelkarp) [inline Makefile]: Like in https://github.com/firecracker-microvm/firecracker-go-sdk/pull/87, a stamp file and picking a better image name would be advisable.

File: sandbox/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,12 @@
+FROM golang:1.12
+
+# Install system dependecies
+#
+RUN apt-get update \
+    && apt-get -y install --no-install-recommends \
+        make
+
+ARG DIR=$(pwd)
+ADD . /firectl
```
Comment (samuelkarp) [inline sandbox/Dockerfile]: Please use `COPY` instead of `ADD`.

File: sandbox/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,12 @@
+FROM golang:1.12
+
+# Install system dependecies
+#
+RUN apt-get update \
+    && apt-get -y install --no-install-recommends \
+        make
+
+ARG DIR=$(pwd)
```
Comment (samuelkarp) [inline sandbox/Dockerfile]: Why do you have it as `ARG`?

File: sandbox/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+ADD . /firectl
+WORKDIR /firectl
+CMD TRIMPATH="$DIR" make release && cp firectl /release
```
Comment (samuelkarp) [inline sandbox/Dockerfile]: I think this would be better: `CMD TRIMPATH=$(pwd) make release && cp firectl /release`

File: Makefile [hunk 8]
```diff
@@ -15,8 +15,28 @@ SRCFILES := *.go
...
+	docker build -f sandbox/Dockerfile -t "firectl" .
+
+sandbox-release-run:
+	docker run \
+		--init \
+		--rm \
+		-v ${MOUNTPATH}:/release \
+		-t firectl
```
Comment (samuelkarp) [inline Makefile]: Why do you have `-t` here?  We don't need a TTY in this container.

**Comment by xibz** [comment]
Going to go ahead and close this PR due to no longer needing trimpath.
